mono/tests/delegate7.cs

```diff
@@ -14,45 +12,114 @@ class Tests {
 		v += 2;
 		Console.WriteLine ("Test.F2");
 	}
-	static void F3 () {
+	static void F4 () {
 		v += 4;
-		Console.WriteLine ("Test.F3");
+		Console.WriteLine ("Test.F4");
 	}
 
 	public static int Main () {
 		return TestDriver.RunTests (typeof (Tests));
 	}
 
+	static int v = 0;
+	static bool check_is_expected_v (SimpleDelegate d, int expected_v)
+	{
+		v = 0;
+		d ();
+		return v == expected_v;
+	}
+
 	static public int test_0_test () {
```

Comment:
@kumpera I added more exhaustive tests for delegate removing.

I also simplified the removing algorithm, as it was also previously wrong